homeassistant/components/palazzetti/__init__.py

```diff
@@ -0,0 +1,40 @@
+"""The Palazzetti integration."""
+
+from __future__ import annotations
+
+from pypalazzetti.exceptions import CommunicationError, ValidationError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+
+from .const import DOMAIN
+from .coordinator import PalazzettiDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [Platform.CLIMATE]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Palazzetti from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    coordinator = PalazzettiDataUpdateCoordinator(hass, entry)
+    try:
+        await coordinator.palazzetti.connect()
+        await coordinator.palazzetti.update_state()
+    except (CommunicationError, ValidationError) as err:
+        raise ConfigEntryError(
+            err, translation_domain=DOMAIN, translation_key="invalid_host"
+        ) from err
```

Comment:
Why would it be an invalid host?

Reply:
The API client lib will throw a `CommunicationError` when the hostname/IP is not responding, or responding something out of spec. Most likely, there isn't a Palazzetti device at this address.

Reply:
But then we should just raise a ConfigEntryNotReady instead so we retry again later